qiskit/circuit/parameterexpression.py

```diff
@@ -306,12 +334,71 @@ def log(self):
         from sympy import log as _log
         return self._call(_log)
 
+    def sqrt(self):
+        """Return square root of ParameterExpression"""
+        from sympy import sqrt as _sqrt
+        return self._call(_sqrt)
+
+    def __floor__(self):
+        """Return floor of ParameterExpression."""
+        from sympy import floor as _floor
+        return self._call(_floor)
+
+    def __ceil__(self):
+        """Return floor of ParameterExpression."""
+        from sympy import ceiling as _ceil
+        return self._call(_ceil)
+
+    def max(self, **kwargs):
+        """Return max expression.
+
+        This is called as numpy.max(expr1, expr2).
+
+        Note: python's max(expr1, expr2) won't work.
+        """
+        from sympy import Max
+        expr1 = self._symbol_expr
+        other = kwargs['axis']  # bit hacky
```

Comment:
When called as an array numpy doesn't go into the ParameterExpression.max but instead calls the class' relational operations and expects boolean results. Similarly, python's max follows the same path expecting a boolean result instead of an expression which would evaluate to boolean on binding.